piptools/sync.py

```diff
@@ -157,26 +164,34 @@ def sync(to_install, to_uninstall, verbose=False, dry_run=False, install_flags=N
     if not verbose:
         pip_flags += ["-q"]
 
-    if to_uninstall:
-        if dry_run:
+    if ask:
+        dry_run = True
+
+    if dry_run:
```

Comment:
```suggestion
    if dry_run or ask:
```
Options `dry_run` and `ask` both prints the intentions to install or uninstall packages, so check them both.

Reply:
I did originally write it this way, but I thought it was slightly nicer to modify the value of `dry_run` for two reasons:
* It makes it clear when reading the code that ask is essentially dry_run, except that with ask we can disable dry_run after a prompt,
* It enables `--ask` to override `--dry-run`. With the changes you suggest, when the user provides both `--ask` and `--dry-run`, it will ask for confirmation, but will still be in dry-run mode so won't do anything after the user says yes.

Reply:
> It enables `--ask` to override `--dry-run`. With the changes you suggest, when the user provides both `--ask` and `--dry-run`, it will ask for confirmation, but will still be in dry-run mode so won't do anything after the user says yes.


Yeah, I've overlooked that case. Thanks for clarifying! But the tests passed. It would be nice to have a test for that case.

Reply:
Good point. I'll add tests for that case.